test/plugins/SetIntegration/set_integration.spec.ts

```diff
@@ -0,0 +1,78 @@
+// @ts-nocheck
+import * as assert from "assert";
+import {
+  NAME,
+  runSetIntegration,
+} from "../../../src/plugins/SetIntegration/set_integration";
+
+describe(NAME, () => {
+  let setLabels: string[];
+  const mockContext = {
+    log: () => undefined,
+    payload: {
+      issue: {},
+    },
+    issue: (val) => val,
+    github: {
+      issues: {
+        async listLabelsForRepo() {
+          return {
+            data: [
+              { name: "integration: awesome" },
+              { name: "integration: awesome_integration" },
+            ],
+          };
+        },
+        async addLabels(labels) {
+          setLabels = labels;
+        },
+      },
+    },
+  };
+
+  beforeEach(function() {
+    setLabels = [];
+  });
+
+  it("Valid integration docLink", async () => {
+    await runSetIntegration({
+      ...mockContext,
+      payload: {
+        issue: {
+          body:
+            "Link to integration documentation on our website: https://www.home-assistant.io/integrations/awesome",
+        },
+      },
+    });
+
+    assert.deepStrictEqual(setLabels, { labels: ["integration: awesome"] });
+  });
+
+  it("Valid integration docLink with underscore", async () => {
+    await runSetIntegration({
+      ...mockContext,
+      payload: {
+        issue: {
+          body:
+            "Link to integration documentation on our website: https://www.home-assistant.io/integrations/awesome_integration",
+        },
+      },
+    });
+
+    assert.deepStrictEqual(setLabels, {
+      labels: ["integration: awesome_integration"],
+    });
+  });
+
+  it("Not valid integration docLink", async () => {
+    await runSetIntegration({
+      ...mockContext,
+      payload: {
+        issue: {
+          body: "https://www.home-assistant.io/integrations/not_valid",
```

Comment:
I'm blind maybe, why is this not valid?

Reply:
Oh because we match the prefix in the description, got it.

Reply:
No, we don't. I don't get it.

Reply:
It's not valid because the integration does not match an existing label

Reply:
Updated descriptions and the body to make that clear

Reply:
It is to prevent new labels are created using random URLs